server-common/src/main/java/org/apache/gravitino/server/authentication/OAuthConfig.java

```diff
@@ -73,4 +73,48 @@ public interface OAuthConfig {
           .stringConf()
           .checkValue(StringUtils::isNotBlank, ConfigConstants.NOT_BLANK_ERROR_MSG)
           .create();
+
+  // Extensible OAuth provider configs
+  ConfigEntry<String> PROVIDER =
+      new ConfigBuilder(OAUTH_CONFIG_PREFIX + "provider")
+          .doc("The OAuth provider to use (e.g., azure)")
+          .version(ConfigConstants.VERSION_0_3_0)
+          .stringConf()
+          .create();
+
+  // Azure-specific configs
+  ConfigEntry<String> AZURE_CLIENT_ID =
+      new ConfigBuilder(OAUTH_CONFIG_PREFIX + "azure.client-id")
```

Comment:
Why do we need `client-id`? I think it may be input by the user.

Reply:
We need client ID for Web UI login. I agree that for machine users like spark-sql we don't need this, but here, we expose client-id through /configs API for front end to use it while creating login

Reply:
If so, Web UI operations will executed by a user. Is it ok?